### PR TITLE
Update max_concurrent_shard_request parameter docs

### PR DESCRIPTION
**Pull request by javanna**
Some of the docs were outdated as they did not mention that the limit is
not per node. Also, The default value changed.

Relates to #31206

### CODE REVIEW

**Comment by javanna** [comment]
run elasticsearch-ci/bwc

File: docs/reference/search/multi-search.asciidoc [hunk 1]
```diff
@@ -86,11 +86,10 @@ the maximum number of concurrent searches the multi search api will execute.
 This default is based on the number of data nodes and the default search thread pool size.
 
 The request parameter `max_concurrent_shard_requests` can be used to control the
-maximum number of concurrent shard requests the each sub search request will execute.
+maximum number of concurrent shard requests the each sub search request will execute per node.
```
Comment (jimczi) [inline docs/reference/search/multi-search.asciidoc]: nit: s/maximum number of concurrent shard requests the/maximum number of concurrent shard requests that/

File: docs/reference/search/multi-search.asciidoc [hunk 2]
```diff
@@ -86,11 +86,10 @@ the maximum number of concurrent searches the multi search api will execute.
...
 This parameter should be used to protect a single request from overloading a cluster
 (e.g., a default request will hit all indices in a cluster which could cause shard request rejections
-if the number of shards per node is high). This default is based on the number of
-data nodes in the cluster but at most `256`.In certain scenarios parallelism isn't achieved through
+if the number of shards per node is high). This default value is `5`.In certain scenarios searcparallelism isn't achieved through
```
Comment (jimczi) [inline docs/reference/search/multi-search.asciidoc]: nit: s/In certain scenarios searcparallelism/In certain scenarios parallelism/

**Review by jimczi — APPROVED** [review]
I left some comments, LGTM otherwise

File: docs/reference/search/multi-search.asciidoc (hunk 2 above)
Comment (javanna) [reply docs/reference/search/multi-search.asciidoc]: bla :) will fix

**Comment by javanna** [comment]
run elasticsearch-ci/bwc

**Comment by javanna** [comment]
run elasticsearch-ci/default-distro

**Pull request merged**